enclave-runner/src/usercalls/mod.rs

```diff
@@ -681,6 +741,21 @@ impl RunningTcs {
         local_addr: Option<&mut OutputBuffer>,
         peer_addr: Option<&mut OutputBuffer>,
     ) -> IoResult<Fd> {
+        let addr_str = String::from_utf8(addr.to_vec());
+        if addr_str.is_ok() {
+            let mut local_addr_str = local_addr.as_ref().map(|_| String::new());
+            let mut peer_addr_str = peer_addr.as_ref().map(|_| String::new());
+            if let Some(stream_ext) = self.enclave.usercall_ext
+                .connect_stream(&addr_str.unwrap(), local_addr_str.as_mut(), peer_addr_str.as_mut())? {
+                if let Some(local_addr) = local_addr {
+                    local_addr.set(local_addr_str.unwrap().into_bytes());
+                }
+                if let Some(peer_addr) = peer_addr {
+                    peer_addr.set(peer_addr_str.unwrap().into_bytes());
+                }
+                return Ok(self.alloc_fd(FileDesc::Stream(stream_ext)));
+            }
+        }
         let addr = str::from_utf8(addr).map_err(|_| IoErrorKind::ConnectionRefused)?;
```

Comment:
We already converted `addr` to utf8 at the beginning of this function, so we shouldn't do it again here. You should be able to just move this line to the top of the function (replacing `let addr_str = ...`)